src/blog/posts/service.py

```diff
@@ -0,0 +1,59 @@
+from blog.db.db import post_table
+from blog.posts.models import PostCreate, PostUpdate, Post
+import typing as t
+
+from blog.utils import max_id
+
+
+def get_all() -> t.List[t.Optional[Post]]:
+    return post_table
+
+
+def get_posts_by_user_id(user_id: int) -> t.List[t.Optional[Post]]:
+    user_posts = [post for post in post_table if post["user_id"] == user_id]
+    return user_posts
+
+
+def get(post_id: int) -> t.Optional[Post]:
+    searched_post = None
+    for post in post_table:
+        if post["id"] == post_id:
+            searched_post = post
+
+    return searched_post
+
+
+def create(post_in: PostCreate) -> Post:
```

Comment:
in create_post view you have a check `if post not None`. This means this exact method should return either Post or None. 
```suggestion
def create(post_in: PostCreate) -> t.Optional[Post]:
```

Reply:
Thank you Polina I did the changes.